.github/workflows/translations.yml

```diff
@@ -1,66 +1,87 @@
 
+    name: Update translation source
     runs-on: ubuntu-latest
 
     steps:
+      - name: Checkout repo
+        uses: actions/checkout@v2
+
       - name: Install lupdate
         shell: bash
         run: |
           sudo apt-get update
           sudo apt-get install -y --no-install-recommends qttools5-dev-tools
 
-      - name: Checkout repo
-        uses: actions/checkout@v2
-
-      - name: Update cockatrice translations
+      - name: Update Cockatrice translation source
+        id: cockatrice
         shell: bash
-        run: |
-          shopt -s globstar # globstar is needed for recursive **
-          lupdate -version
-          echo "reading the following source files:"
-          # note: there are three strings to translate in common right now
-          echo {cockatrice,common}/**/*.{cpp,h}
-          echo "$(echo {cockatrice,common}/**/*.{cpp,h} | wc -w) files total"
-          lupdate {cockatrice,common}/**/*.{cpp,h} -ts cockatrice/translations/cockatrice_en@source.ts
+        env:
+          FILE: 'cockatrice/cockatrice_en@source.ts'
+          DIRS: 'cockatrice/src common'
+        run: .ci/update_translations.sh
 
-      - name: Update oracle translations
+      - name: Update Oracle translation source
+        id: oracle
         shell: bash
-        run: |
-          shopt -s globstar # globstar is needed for recursive **
-          lupdate -version
-          echo "reading the following source files:"
-          echo oracle/**/*.{cpp,h}
-          echo "$(echo oracle/**/*.{cpp,h} | wc -w) files total"
-          lupdate oracle/**/*.{cpp,h} -ts oracle/translations/oracle_en@source.ts
+        env:
+          FILE: 'oracle/oracle_en@source.ts'
+          DIRS: 'oracle/src'
+        run: .ci/update_translations.sh
 
-      - name: Check for updates
-        id: check
-        shell: bash
-        run: |
-          set +e # do not fail, just save the exit state
-          git diff --exit-code
-          echo "::set-output name=deploy::$?"
+      - name: Render template
+        id: template
+        uses: chuhlomin/render-template@v1.4
+        with:
+          template: .ci/update_translations_template.md
+          vars: |
+            cockatrice_output: ${{ steps.cockatrice.outputs.output }}
+            oracle_output: ${{ steps.oracle.outputs.output }}
+            commit: ${{ github.sha }}
+
+      - name: Create pull request
+        if: github.event_name != 'pull_request'
+        id: create_pr
+        uses: peter-evans/create-pull-request@v4
+        with:
+          add-paths: |
+            cockatrice/cockatrice_en@source.ts
+            oracle/oracle_en@source.ts
+          commit-message: Update translation source strings
+          # author is the owner of the commit
+          author: github-actions <github-actions@github.com>
+          branch: ci-update_translations
+          delete-branch: true
+          title: '[Translations] Update source strings'
```

Comment:
Maybe annotate it's automatically generated?

Reply:
the body of the commit contains that and more, the template for that is added here as well